test/e2e/es/version_upgrade_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package es
+
+import (
+	"testing"
+
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
+)
+
+func TestVersionUpgrade680To720(t *testing.T) {
+	initial := elasticsearch.NewBuilder("test-version-up-680-to-720").
+		WithVersion("6.8.0").
+		WithESMasterDataNodes(3, elasticsearch.DefaultResources)
+
+	mutated := initial.WithNoESTopology().
+		WithVersion("7.2.0").
+		WithESMasterDataNodes(3, elasticsearch.DefaultResources)
+
+	RunESMutation(t, initial, mutated)
+}
+
+func TestVersionUpgradeSingle680To720(t *testing.T) {
+	initial := elasticsearch.NewBuilder("test-version-up-680-to-720").
```

Comment:
Maybe name the test `test-version-up-1-680-to-720` (we can go up to 36-1-4=31 chars) so that the test names remain unique.